tirocks/src/db/open_options.rs

```diff
@@ -0,0 +1,347 @@
+            panic!("ttl is missing for {}", name);
+        }
+        self.cfs.push((name, cf));
+        self
+    }
+
+    #[inline]
+    pub fn db_options_mut(&mut self) -> &mut DbOptions {
+        &mut self.db
+    }
+
+    /// A shortcut for the common used `db_options_mut().set_create_if_missing()`.
+    #[inline]
+    pub fn set_create_if_missing(&mut self, create: bool) -> &mut Self {
+        self.db.set_create_if_missing(create);
+        self
+    }
+
+    /// A shortcut for `db_options_mut().set_create_missing_column_families()`.
+    #[inline]
+    pub fn set_create_missing_column_families(&mut self, create: bool) -> &mut Self {
+        self.db.set_create_missing_column_families(create);
+        self
+    }
+
+    pub fn set_read_only(&mut self, enable: bool, error_if_log_exists: bool) -> &mut Self {
+        self.error_if_log_exists = if enable {
+            Some(error_if_log_exists)
+        } else {
+            None
+        };
+        self
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct MultiCfTitanBuilder {
+    db: TitanDbOptions,
+    cfs: Vec<(String, TitanCfOptions)>,
+}
+
+impl MultiCfTitanBuilder {
+    pub fn new(db: TitanDbOptions) -> MultiCfTitanBuilder {
+        MultiCfTitanBuilder { db, cfs: vec![] }
+    }
+
+    pub fn add_cf(&mut self, name: impl Into<String>, cf: TitanCfOptions) -> &mut Self {
+        self.cfs.push((name.into(), cf));
+        self
+    }
+
+    #[inline]
+    pub fn db_options_mut(&mut self) -> &mut TitanDbOptions {
+        &mut self.db
+    }
+
+    /// A shortcut for the common used `db_options_mut().set_create_if_missing()`.
+    #[inline]
+    pub fn set_create_if_missing(&mut self, create: bool) -> &mut Self {
+        self.db.set_create_if_missing(create);
+        self
+    }
+
+    /// A shortcut for `db_options_mut().set_create_missing_column_families()`.
+    #[inline]
+    pub fn set_create_missing_column_families(&mut self, create: bool) -> &mut Self {
+        self.db.set_create_missing_column_families(create);
+        self
+    }
+}
+
+pub trait OpenOptions {
+    /// Open the db and check if it's titan.
+    ///
+    /// # Safety
+    ///
+    /// Opened cf will be pushed to `handles`. And they must be used while db pointer
+    /// is still alive.
+    unsafe fn open_raw(
+        &self,
+        comparator: &mut Vec<Arc<SysComparator>>,
+        env: &mut Option<Arc<Env>>,
+        handles: &mut Vec<*mut rocksdb_ColumnFamilyHandle>,
+        path: &Path,
+    ) -> Result<(*mut rocksdb_DB, bool)>;
+
+    fn open(&self, path: &Path) -> Result<Db> {
+        let mut comparator = vec![];
+        let mut env = None;
+        let mut handles = Vec::with_capacity(0);
+        let (ptr, is_titan) =
+            unsafe { self.open_raw(&mut comparator, &mut env, &mut handles, path) }?;
+        let handles = if handles.is_empty() {
+            vec![unsafe {
+                let ptr = tirocks_sys::crocksdb_get_default_column_family(ptr);
```

Comment:
If I remember correctly, you can't use this handle as if it's created from `crocksdb_open_column_families`. i.e. you can't destroy it.